categorize_notify/program.py

```diff
@@ -9,64 +9,106 @@
 """
 
 import base64
+from datetime import datetime, timezone
+import os
 import time
 
 import autokitteh
 from autokitteh import google, openai, slack
 
 
+POLL_INTERVAL = os.getenv("POLL_INTERVAL")
 SLACK_CHANNELS = ["demos", "engineering", "ui"]
+gmail = google.gmail_client("my_gmail").users()
+processed_message_ids = set()
+start_time = datetime.now(timezone.utc).timestamp()
 
 
 def on_http_get(event):
-    total_messages = None
     while True:
-        total_messages = _poll_inbox(total_messages)
-        time.sleep(10)
+        _poll_inbox()
+        time.sleep(float(POLL_INTERVAL))
+
+
+def _poll_inbox():
+    token = None
+    current_message_ids = set()
+
+    while True:
+        results = get_new_inbox_messages(token)
+
+        current_message_ids.update({msg["id"] for msg in results.get("messages", [])})
+
+        token = results.get("nextPageToken")
+        if not token:
+            break
+
+    new_message_ids = current_message_ids - processed_message_ids
+
+    for message_id in new_message_ids:
+        _process_email(message_id, start_time)
+
+    processed_message_ids.update(new_message_ids)
```

Comment:
Are you **sure** it works without specifying that `processed_message_ids` is `global`? It should be required, but even if not it's a good self-documenting clarification.

```suggestion
    global processed_message_ids
    processed_message_ids.update(new_message_ids)
```

Reply:
>In Python, the global keyword allows you to change a variable value outside of its current scope. It is used to make changes to a global variable from a local location. The global keyword is only required for altering the variable value and not for publishing or accessing it. [source](https://www.toppr.com/guides/python-guide/references/methods-and-functions/global-keyword/python-global-keyword-with-examples/)

>Minimize Use: Only use global variables when necessary. [source](https://lifewithdata.com/2023/10/20/python-global-keyword/)

Because the set isn't being reassigned to another set or something else entirely, global is not required. I'm not 100% but I don't think it would be considered best practice.